internal/utils_test.go

```diff
@@ -93,11 +95,37 @@ func TestXSyncMapCounterMap(t *testing.T) {
 			go func() {
 				defer wg.Done()
 				cm.Inc("key")
+				cm.Inc("key2")
 			}()
 		}
 		wg.Wait()
 
-		assert.Equal(map[string]int64{"key": 10}, cm.GetAndReset())
+		assert.Equal(map[string]int64{"key": 10, "key2": 10}, cm.GetAndReset())
+	})
+
+	t.Run("concurrent with reset", func(t *testing.T) {
+		assert := assert.New(t)
+		cm := NewXSyncMapCounterMap()
+		wg := sync.WaitGroup{}
+		var val atomic.Int64
+		for range 10 {
+			wg.Add(2)
```

Comment:
By spawning, two goroutines in this fashion, I don't think you can guarantee you collect all the results. You might run the `GetAndReset` before you increment,